flake.nix

```
@@ -0,0 +1,79 @@
+{
+  description = "webR development environment";
+
+  # Flake inputs
+  inputs = {
+    nixpkgs.url = "https://flakehub.com/f/NixOS/nixpkgs/0.2305.492102.tar.gz";
+  };
+
+  # Flake outputs
+  outputs = { self, nixpkgs }:
+    let
+      # Systems supported
+      allSystems = [
+        "x86_64-linux" # 64-bit Intel/AMD Linux
+        "aarch64-linux" # 64-bit ARM Linux
+        "x86_64-darwin" # 64-bit Intel macOS
+        "aarch64-darwin" # 64-bit ARM macOS
+      ];
+
+      # Helper to provide system-specific attributes
+      forAllSystems = f: nixpkgs.lib.genAttrs allSystems (system: f {
+        pkgs = import nixpkgs { inherit system; };
+      });
+
+    in
+    {
+      # Development environment output
+      devShells = forAllSystems ({ pkgs }: {
+        default = pkgs.mkShell {
+          # The Nix packages provided in the environment
+          packages = with pkgs; [
+            emscripten
+            cmake
+            gperf
+            lzma
+            pcre2
+            nodejs_18
+            quilt
+            wget
+
+            libxml2
+            git
+            python3
+
+            # Inputs for building R borrowed from:
+            # https://github.com/NixOS/nixpkgs/blob/85f1ba3e/pkgs/applications/science/math/R/default.nix
+            bzip2
+            gfortran
+            perl
+            xz
+            zlib
+            icu
+            bison
+            which
+            blas
+            lapack
+            curl
+            tzdata
+
+            pkg-config  # For fontconfig
+            sqlite  # For proj
```

I needed to add the following for `make all` in the `libs` directory.

```suggestion
            sqlite  # For proj
            glib # For pango
            unzip # For extracting font data
```

Moving on to building LLVM flang and webR now...

---

> Moving on to building LLVM flang and webR now...

Update: This seems to have been successful, so I think this covers all the packages we need (other than the process of extracting `flang` into its own flake).